docs/frontend-system/architecture/05-extension-overrides.md

```diff
@@ -10,38 +10,149 @@ description: Frontend extension overrides
 
-Talk about how extension overrides
+## Override App Extensions
 
--->
+In order to override an app extension, you must create a new extension and add it to the list of overridden features. The steps are: create your extension overrides and use them in Backstage.
 
-## Creating a Extension Override
+### Example
 
-<!--
+In the example below, we create a file that exports custom extensions for the app's `light` and `dark` themes:
 
-How to create an extension override and export it from a package.A
+```tsx title="packages/app/src/themes.ts" {10-11,22-23} showLineNumbers
+import {
+  createThemeExtension,
+  createExtensionOverrides
+} from '@backstage/frontend-plugin-api';
+import { apertureThemes } from './themes';
+import { ApertureLightIcon, ApertureDarkIcon } from './icons';
 
- - Example using the default exports
+// Creating a light theme extension
+const apertureLightTheme = createThemeExtension({
+  namespace: 'app',
+  name: 'light',
+  title: 'Aperture Light Theme',
+  variant: 'light',
+  icon: <ApertureLightIcon />,
+  Provider: ({ children }) => (
+    <UnifiedThemeProvider theme={apertureThemes.light} children={children} />
+  ),
+});
 
-You can also create them directly withing the app for local overrides.
+// Creating a dark theme extension
+const apertureDarkTheme = createThemeExtension({
+  namespace: 'app',
+  name: 'dark',
+  title: 'Aperture Dark Theme',
+  variant: 'dark',
+  icon: <ApertureDarkIcon />,
+  Provider: ({ children }) => (
+    <UnifiedThemeProvider theme={apertureThemes.dark} children={children} />
+  ),
+});
 
- - Example of installing an override directly into an app
+// Creating an extension overrides preset
+export createExtensionOverrides({
+  extensions: [apertureLightTheme, apertureDarkTheme]
+});
+```
 
-Mention that in can still be a good idea to split your overrides out into separate packages in large projects. But it's up to you to decide how to group the extensions into extension overrides.
+Lines `10` and `22` should be highlighted because they declare `namespace` as `'app'`, so the system knows we are overriding app extensions. Additionally, to specifically override the `light` and `dark` theme extensions, we set the `name` option to `light` and `dark` respectively on lines `11` and `23`. Therefore, to override app theme extensions, we ensure that the extension `namespace` and `name` match those of the default app theme extension definitions.
```

Comment:
Line numbers are tricky to keep in sync, I think it would be best to avoid using the and instead refer to pieces of the example through text or `[N]` markers

Reply:
Done